dkc/core/rest/folder.py

```diff
@@ -182,7 +182,8 @@ def perform_create(self, serializer: FolderSerializer):
             if not tree.has_permission(user, permission=Permission.write):
                 raise PermissionDenied()
         else:
-            tree = Tree.objects.create()
+            # Use the user's quota for the new tree
+            tree = Tree.objects.create(quota=user.quota)
```

Comment:
Just a note at how powerful this particular line is. I'm very pleased at how easy this was.